engine/common/synchronization/engine_spam_test.go

```diff
@@ -0,0 +1,259 @@
+				require.NoError(ss.T(), ss.e.Process(channels.SyncCommittee, originID, req))
+			}
+
+			// check function call expectations at the end of the load test; otherwise, load test would take much longer
+			ss.core.AssertExpectations(ss.T())
+			ss.con.AssertExpectations(ss.T())
+
+			// check that correct range of misbehavior reports were generated
+			// since we're using a probabilistic approach to generate misbehavior reports, we can't guarantee the exact number,
+			// so we check that it's within an expected range
+			ss.T().Logf("misbehaviors counter after load test: %d (expected lower bound: %d expected upper bound: %d)", misbehaviorsCounter, loadGroup.expectedMisbehaviorsLower, loadGroup.expectedMisbehaviorsUpper)
+			assert.GreaterOrEqual(ss.T(), misbehaviorsCounter, loadGroup.expectedMisbehaviorsLower)
+			assert.LessOrEqual(ss.T(), misbehaviorsCounter, loadGroup.expectedMisbehaviorsUpper)
+
+			misbehaviorsCounter = 0 // reset counter for next subtest
+		})
+	}
+}
+
+// TestLoad_Process_RangeRequest_SometimesReportSpam is a load test that ensures that a misbehavior report is generated
+// an appropriate range of times when the base probability factor and range are set to different values.
+func (ss *SyncSuite) TestLoad_Process_RangeRequest_SometimesReportSpam() {
+	ctx, cancel := irrecoverable.NewMockSignalerContextWithCancel(ss.T(), context.Background())
+	ss.e.Start(ctx)
+	unittest.AssertClosesBefore(ss.T(), ss.e.Ready(), time.Second)
+	defer cancel()
+
+	load := 1000
+
+	type loadGroup struct {
+		rangeRequestBaseProb      float32
+		expectedMisbehaviorsLower int
+		expectedMisbehaviorsUpper int
+		fromHeight                uint64
+		toHeight                  uint64
+	}
+
+	loadGroups := []loadGroup{}
+
+	// using a very small range (1) with a 10% base probability factor, expect to almost never get misbehavior report, about 0.003% of the time (3 in 1000 requests)
+	// expected probability factor: 0.1 * ((10-9) + 1)/64 = 0.003125
+	loadGroups = append(loadGroups, loadGroup{0.1, 0, 15, 9, 10})
+
+	// using a small range (10) with a 10% base probability factor, expect to get misbehavior report about 1.7% of the time (17 in 1000 requests)
+	// expected probability factor: 0.1 * ((11-1) + 1)/64 = 0.0171875
+	loadGroups = append(loadGroups, loadGroup{0.1, 5, 31, 1, 11})
+
+	// using a large range (99) with a 10% base probability factor, expect to get misbehavior report about 15% of the time (150 in 1000 requests)
+	// expected probability factor: 0.1 * ((100-1) + 1)/64 = 0.15625
+	loadGroups = append(loadGroups, loadGroup{0.1, 110, 200, 1, 100})
+
+	// using a flat range (0) (from height == to height) with a 1% base probability factor, expect to almost never get a misbehavior report, about 0.16% of the time (2 in 1000 requests)
+	// expected probability factor: 0.01 * ((1-1) + 1)/64 = 0.0015625
+	loadGroups = append(loadGroups, loadGroup{0.01, 0, 5, 1, 1})
```

Comment:
```suggestion
	loadGroups = append(loadGroups, loadGroup{0.01, 0, 0, 1, 1})
```
Shouldn't the `expectedMisbehaviorsUpper` be also `0` in this case?

Reply:
Because of the probabilistic nature of the load tests, you sometimes get edge cases (from my extensive local testing) that cover outliers where out of a 1000 messages, up to 5 could be reported as spam. 5/1000 = 0.005 and the calculated probability is 0.00171875 which 2.9x as small. So this is to avoid flakiness in CI.